src/status_im2/common/nickname_drawer/view.cljs

```diff
@@ -0,0 +1,84 @@
+(ns status-im2.common.nickname-drawer.view
+  (:require [clojure.string :as string]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.icon :as icons]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.common.nickname-drawer.style :as style]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn empty-nickname?
+  [nickname]
+  (string/blank? @nickname))
+
+(defn add-nickname-toast
+  [primary-name entered-nickname public-key]
+  (fn []
+    (rf/dispatch [:hide-bottom-sheet])
+    (rf/dispatch [:toasts/upsert
+                  {:id         :add-nickname
+                   :icon       :i/correct
+                   :icon-color (colors/theme-colors colors/success-60 colors/success-50)
+                   :text       (i18n/label
+                                :t/set-nickname-toast
+                                {:primary-name primary-name
+                                 :nickname     (string/trim @entered-nickname)})}])
```

Comment:
hmm, should we not pass the atom through as a derefed atom. 
i.e:
```
:nickname (string/trim entered-nickname)})}])
```

and so the function is called like: 
```
[add-nickname-toast "some name" @some-atom "public-key"]
```
Would be better if this function didn't need to know if it's value is an atom or not.

Reply:
I agree .. Fixing